client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -78,31 +101,54 @@ export function SitesDashboard( { queryParams }: SitesDashboardProps ) {
 				</HeaderControls>
 			</PageHeader>
 			<PageBodyWrapper>
-				<ClassNames>
-					{ ( { css } ) => (
-						<SitesTableFilterTabs
-							allSites={ sites }
-							className={ css`
-								${ wideCentered }
-								position: relative;
-								top: -48px;
-							` }
-							filterOptions={ queryParams }
-						>
-							{ ( filteredSites, filterOptions ) =>
-								filteredSites.length ? (
-									<SearchableSitesTable
-										sites={ filteredSites }
-										initialSearch={ queryParams.search }
+				<SitesTableFilterTabs
+					tabs={ tabs }
+					initialTabName={ selectedTabName }
+					onSelect={ handleTabSelect }
+				>
+					{ () =>
+						selectedTabHasSites ? (
+							<>
+								<SearchWrapper>
+									<SitesSearch
+										searchIcon={ <SitesSearchIcon /> }
+										onSearch={ handleSearch }
+										isReskinned
+										placeholder={ __( 'Search by name or domain…' ) }
+										defaultValue={ search }
 									/>
+								</SearchWrapper>
+								{ filteredSites.length > 0 ? (
+									<SitesTable sites={ filteredSites } />
 								) : (
-									<NoSitesMessage status={ filterOptions.status } />
-								)
-							}
-						</SitesTableFilterTabs>
-					) }
-				</ClassNames>
+									<h2>{ __( 'No sites match your search.' ) }</h2>
+								) }
+							</>
+						) : (
+							<NoSitesMessage status={ selectedTabName } />
+						)
+					}
+				</SitesTableFilterTabs>
 			</PageBodyWrapper>
 		</main>
 	);
 }
+
+function handleTabSelect( tabName: string ) {
+	page(
+		'all' === tabName
+			? removeQueryArgs( window.location.pathname + window.location.search, 'status' )
+			: addQueryArgs( window.location.pathname + window.location.search, { status: tabName } )
+	);
+}
+
+function handleSearch( rawTerm: string ) {
+	const trimmedTerm = rawTerm.trim();
+	if ( trimmedTerm.length ) {
+		page(
+			addQueryArgs( window.location.pathname + window.location.search, { search: trimmedTerm } )
+		);
+	} else {
+		page( removeQueryArgs( window.location.pathname + window.location.search, 'search' ) );
+	}
+}
```

Comment:
We're adding more and more attributes here. Does it make sense to create an abstraction on top of it that would manage the query state?

Reply:
Do you think maybe something like a `handleQueryParamChange()` function that could do the underlying work? e.g.

```js
function handleTabSelect( tabName: string ) {
	const status = tabName === 'all' ? undefined : tabName;
	handleQueryParamChange( { status } );
}

function handleSearch( rawTerm: string ) {
	handleQueryParamChange( { search: term.trim() } );
}

function handleQueryParamChange( { search, status }: { search?: string, status?: string } ) {
	// ...
}
```

Reply:
Something like that! I thought the signature like that so it scales and the size of the object parameter doesn't grow:

```ts
function handleQueryParamChange( param: 'union' | 'of' | 'params', value: string | null )
```

Reply:
Follow up PR https://github.com/Automattic/wp-calypso/pull/65875